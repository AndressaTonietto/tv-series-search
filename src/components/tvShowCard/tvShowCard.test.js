import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";

import { BrowserRouter as Router } from "react-router-dom";
import TvShowCard from "./tvShowCard";
import renderer from "react-test-renderer";

const testShow = {
  id: 1,
  name: "The IT Crowd",
  rating: { average: 8.7 },
  image: { medium: "test-image-url" },
  officialSite: "https://guthib.com/",
};

describe("TvShowCard", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Router>
          <TvShowCard show={testShow} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the tv show general info and watch button correctly", () => {
    render(
      <Router>
        <TvShowCard show={testShow} />
      </Router>
    );

    expect(screen.getByText(testShow.name)).toBeInTheDocument();
    expect(
      screen.getByText(testShow.rating.average.toString())
    ).toBeInTheDocument();
    expect(screen.getByAltText(`${testShow.name} poster`)).toHaveAttribute(
      "src",
      testShow.image.medium
    );
    expect(screen.getByText("Watch")).toHaveAttribute(
      "href",
      testShow.officialSite
    );
  });
});
