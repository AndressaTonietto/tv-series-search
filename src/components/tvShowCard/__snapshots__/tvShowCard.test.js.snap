// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvShowCard renders correctly 1`] = `
<div
  className="card card-compact min-h-56 card-side bg-base-100 shadow-xl cursor-pointer md:w-1/2 lg:w-1/3 xl:w-1/4"
  onClick={[Function]}
>
  <figure>
    <img
      alt="The IT Crowd poster"
      className="max-w-52"
      src="test-image-url"
    />
  </figure>
  <div
    className="card-body"
  >
    <h2
      className="card-title"
    >
      The IT Crowd
    </h2>
    <p>
      8.7
    </p>
    <div
      className="card-actions justify-end"
      onClick={[Function]}
    >
      <a
        className="btn btn-primary false"
        href="https://guthib.com/"
        target="_blank"
      >
        Watch
      </a>
    </div>
  </div>
</div>
`;
